src/frontend/Ast_to_Mir.ml

```diff
@@ -255,18 +267,78 @@ let extract_constraint_args = function
    |CholeskyCov | Correlation | Covariance | Identity ->
       []
 
-let rec gen_check decl_type decl_id decl_trans smeta adlevel =
+let rec base_type = function
+  | SArray (t, _) -> base_type t
+  | SVector _ | SRowVector _ | SMatrix _ -> UReal
+  | x -> remove_size x
+
+let internal_of_dread = function
+  | ReadParam -> FnReadParam
+  | ReadData -> FnReadData
+
+let assign_indexed vident smeta varfn var =
+  let indices =
+    match var.expr with Indexed (_, indices) -> indices | _ -> []
```

Comment:
Do you want to throw here if `var.expr` is not `Indexed (_. _)` or `Var _`?

Reply:
I don't think so, why do you ask?